include/xen/be/XenGnttab.hpp

```diff
@@ -92,29 +92,33 @@ class XenGnttabBuffer
 public:
 
 	/**
-	 * @param[in] domId domain id
-	 * @param[in] ref   grant reference id
-	 * @param[in] prot  same flag as in mmap()
+	 * @param[in] domId  domain id
+	 * @param[in] ref    grant reference id
+	 * @param[in] prot   same flag as in mmap()
+	 * @param[in] offset offset of the data inside the buffer
 	 */
 	XenGnttabBuffer(domid_t domId, grant_ref_t ref,
-					int prot = PROT_READ | PROT_WRITE);
+					int prot = PROT_READ | PROT_WRITE,
+					size_t offset = 0);
 
 	/**
-	 * @param[in] domId domain id
-	 * @param[in] refs  array of grant reference ids
-	 * @param[in] count number of grant refgerence ids
-	 * @param[in] prot  same flag as in mmap()
+	 * @param[in] domId  domain id
+	 * @param[in] refs   array of grant reference ids
+	 * @param[in] count  number of grant refgerence ids
+	 * @param[in] prot   same flag as in mmap()
+	 * @param[in] offset offset of the data inside the buffer
 	 */
 	XenGnttabBuffer(domid_t domId, const grant_ref_t* refs, size_t count,
-					int prot = PROT_READ | PROT_WRITE);
+					int prot = PROT_READ | PROT_WRITE,
+					size_t offset = 0);
 	XenGnttabBuffer(const XenGnttabBuffer&) = delete;
 	XenGnttabBuffer& operator=(XenGnttabBuffer const&) = delete;
 	~XenGnttabBuffer();
 
 	/**
 	 * Returns pointer to the mapped buffer.
 	 */
-	void* get() const { return mBuffer; }
+	void* get() const { return static_cast<char*>(mBuffer) + mOffset; }
```

Comment:
Maybe it is good point to use uint8_t instead of char (since this is not a character operation).

Reply:
hm, probably. will fix